src/CoreAnimation/CATextLayer.cs

```diff
@@ -134,6 +134,52 @@ public void SetFont (NSFont font)
 				}
 			}
 		}
+#if !XAMCORE_4_0
+		[Advice ("Use 'TextTruncationMode' instead.")]
+		public string TruncationMode {
+			get {
+				return NSString.FromHandle (_TruncationMode);
+			}
+			set {
+				if (value == null)
+					throw new ArgumentNullException (nameof (value));
+				var nsvalue = NSString.CreateNative (value);
+				_TruncationMode = nsvalue;
+				NSString.ReleaseNative (nsvalue);
+			}
+		}
+
+		[Advice ("Use 'TextAlignmentMode' instead.")]
+		public string AlignmentMode {
+			get {
+				return NSString.FromHandle (_AlignmentMode);
+			}
+			set {
+				if (value == null)
+					throw new ArgumentNullException (nameof (value));
+				var nsvalue = NSString.CreateNative (value);
+				_AlignmentMode = nsvalue;
+				NSString.ReleaseNative (nsvalue);
+			}
+		}
+#endif // !XAMCORE_4_0
+		public CATextLayerTruncationMode TextTruncationMode {
+			get {
+				return CATextLayerTruncationModeExtensions.GetValue (Runtime.GetNSObject<NSString> (_TruncationMode));
+			}
+			set {
+				_TruncationMode = value.GetConstant ().Handle;
```

Comment:
`value.GetConstant ()` could be `null`, e.g. if the constant does not exists in the executing OS version, leading to an NRE
why not have `_TruncationMode` be an `NSString` and let it deal with `null` ?